### PR TITLE
ci(release): Use action-prepare-release and rework dry-run

### PR DESCRIPTION
**Pull request by BYK**
Switches to using getsentry/action-prepare-release for common steps and removes the dry run option from the prepare stage as it prevents us from testing the publish step. Prepare step should be safe anyways and can be skipped with the `skip-prepare` input.

Related: https://app.asana.com/0/1198192131329257/1198192131329301, https://app.asana.com/0/1198192131329257/1198192573081717

### CODE REVIEW

**Review by chadwhitacre — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -13,57 +13,40 @@ on:
...
-          env:
-            DRY_RUN: ${{ github.event.inputs.dry_run }}
-            GITHUB_API_TOKEN: ${{ secrets.GH_SENTRY_BOT_PAT }}
-            DOCKER_USERNAME: 'sentrybuilder'
-            DOCKER_PASSWORD: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Prepare release
+        uses: getsentry/action-prepare-release@main
+        with:
+          version: ${{ github.event.inputs.version }}
+          force: ${{ github.event.inputs.force }}
```
Comment (chadwhitacre) [inline .github/workflows/release.yml:32]: Did we want to standardize on `INPUT_FOO`?

Comment (BYK) [reply .github/workflows/release.yml:32]: They are different. When using in a bash script, I prefer `$INPUT_FOO`, everywhere else, have to use the GitHub Action expression syntax.

**Review by chadwhitacre — APPROVED** [review]
https://github.com/getsentry/snuba/runs/1494665431?check_suite_focus=true

👍

**Comment by BYK** [comment]
Snuba team should not be a codeowner for this file now but I guess GitHub doesn't update old PRs.

@fpacifici would appreciate a stamp

**Review by fpacifici — APPROVED** [review]

**Pull request merged**